backend/app/middlewares/verifyMenuItem.js

```diff
@@ -0,0 +1,36 @@
+const db = require("../models");
+const CATEGORIES = db.CATEGORIES;
+const MenuItem = db.menuitem;
+
+checkDuplicateMealName = (req, res, next) => {
+  MenuItem.findOne({
+    name: req.body.name,
+  }).exec((err, menuitem) => {
+    if (err) {
+      res.status(500).send({ message: err });
+      return;
+    }
+
+    if (menuitem) {
+      res.status(400).send({ message: "Failed! Meal name already exists!" });
+      return;
+    }
+
+    next();
+  });
+};
+
+checkCategoriesExist = (req, res, next) => {
+  if (req.body.categories) {
```

Comment:
Shouldn't you be checking that `req.body.category` exists in the `Category` collection in your database instead?